ycmd/completers/cs/cs_completer.py

```diff
@@ -237,7 +237,10 @@ def _StartServer( self, request_data ):
       raise RuntimeError( 'Autodetection of solution file failed.\n' )
     self._logger.info( u'Loading solution file {0}'.format( path_to_solutionfile ) )
 
-    self._omnisharp_port = utils.GetUnusedLocalhostPort()
+    self._omnisharp_port = int( self.user_options.get( 'csharp_server_port', 0 ) )
```

Comment:
Could you extract this small bit of port-selection logic into a helper method? `_StartServer` is already pretty big.